Sources/SwiftSyntaxBuilder/Syntax+StringInterpolation.swift

```diff
@@ -160,3 +242,197 @@ extension SyntaxExpressibleByStringInterpolation {
     try self.init(stringInterpolationOrThrow: interpolation)
   }
 }
+
+// MARK: ExpressibleByLiteralSyntax conformances
+
+extension Substring: ExpressibleByLiteralSyntax {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    String(self).makeLiteralSyntax()
+  }
+}
+
+extension String: ExpressibleByLiteralSyntax {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    // TODO: Use a multi-line literal if there are more than N inner newlines.
+    StringLiteralExpr(content: self)
+  }
+}
+
+extension ExpressibleByLiteralSyntax where Self: BinaryInteger {
+  public func makeLiteralSyntax() -> ExprSyntaxProtocol {
+    // TODO: Radix selection? Thousands separators?
```

Comment:
Personally, I think we can just remove this `TODO`. I can’t think of any situation where we could intelligently pick a radix other than 10 and thousands separators are also up to personal style and I don’t think we should assume that clients want them.

(same below for float literals)